src/main/java/org/prebid/server/bidder/deepintent/DeepintentBidder.java

```diff
@@ -69,16 +72,35 @@ public DeepintentBidder(String endpointUrl, JacksonMapper mapper) {
         return Result.of(requests, errors);
     }
 
+    private Banner buildImpBanner(Banner banner, String impId) {
+        if (banner == null) {
+            throw new PreBidException(String.format("We need a Banner Object in "
+                    + "the request, imp : %s", impId));
+        }
+
+        if (banner.getW() == null && banner.getH() == null) {
+            final List<Format> bannerFormats = banner.getFormat();
+            if (CollectionUtils.isEmpty(banner.getFormat())) {
+                throw new PreBidException(String.format("At least one size is required, imp : %s", impId));
+            }
+            final Format format = bannerFormats.get(0);
+            return banner.toBuilder().w(format.getW()).h(format.getH()).build();
+        }
+
+        return banner;
+    }
+
     private ExtImpDeepintent parseImpExt(Imp imp) {
         try {
             return mapper.mapper().convertValue(imp.getExt(), DEEPINTENT_EXT_TYPE_REFERENCE).getBidder();
         } catch (IllegalArgumentException e) {
-            throw new PreBidException(e.getMessage());
+            throw new PreBidException(String.format("Impression id=%s, has invalid Ext", imp.getId()));
         }
     }
 
-    private Imp modifyImp(Imp imp, String tagId) {
+    private Imp modifyImp(Imp imp, String tagId, Banner updatedBanner) {
```

Comment:
`updatedBanner` can be renamed to just `banner` since the method doesn't know the context.